src/protocols/secure_channel/MessageCounterManager.cpp

```diff
@@ -101,6 +101,15 @@ CHIP_ERROR MessageCounterManager::OnMessageReceived(Messaging::ExchangeContext *
 
 void MessageCounterManager::OnResponseTimeout(Messaging::ExchangeContext * exchangeContext)
 {
+    if (mExchangeMgr == nullptr)
+    {
+        if (exchangeContext != nullptr)
```

Comment:
can exchangeContext be nullptr? This feels a bit special cased. If it can be null, I believe we should have the same check below where we call close.

Reply:
I don't think it can be null, I can try to use a reference here to ensure that it can't be null.

Reply:
exchangeContext can't be null here.

Reply:
I remove the test of `exchangeContext`.